rita/src/rita_common/network_endpoints/mod.rs

```diff
@@ -44,52 +47,56 @@ pub fn make_payments(
     PaymentController::from_registry()
         .send(rita_common::payment_controller::PaymentReceived(
             pmt.0.clone(),
-        ))
-        .from_err()
+        )).from_err()
         .and_then(|_| Ok(HttpResponse::Ok().into()))
         .responder()
 }
 
 pub fn hello_response(
     req: (Json<serde_json::Value>, HttpRequest),
 ) -> Box<Future<Item = Json<LocalIdentity>, Error = Error>> {
-    let new_id: Result<Identity, serde_json::Error> = serde_json::from_value(req.0.clone());
-    let old_id: Result<LocalIdentity, serde_json::Error> = serde_json::from_value(req.0.clone());
-
-    // remove in Alpha 6
-    let their_id = match new_id {
-        Ok(new_id) => {
-            info!("got new id sending new response");
-            new_id
+    let id: Result<LocalIdentity, serde_json::Error> = serde_json::from_value(req.0.clone());
+
+    let their_id = match id {
```

Comment:
We should just be able to make req take a Json<LocalIdentity> instead of a serde value, should make this code cleaner

Reply:
so like Json<Hello>?

Reply:
sorry not really seeing the syntax for this. Don't we have to parse?

Reply:
do something like  req: (Json<LocalIdentity>, HttpRequest)